packages/nest-api/src/scraping-processor-service/keyword-scrape.service.spec.ts

```diff
@@ -0,0 +1,231 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import * as puppeteer from 'puppeteer';
+import { KeywordScrapeService } from './keyword-scrape.service';
+import { ClientProxy, RmqRecordBuilder } from '@nestjs/microservices';
+import { SCRAPING_DONE } from '../core/constants';
+import { ScrapeJobDonePayload, ScrapeKeywordPayload } from '../core/types';
+import { GooglePageSelectors, RmqMessagePatterns } from '../core/enums';
+import { Page } from 'puppeteer';
+
+jest.mock('puppeteer');
+
+describe('KeywordScrapeService', () => {
+  let service: KeywordScrapeService;
+  let mockRmqClient: ClientProxy;
+
+  beforeEach(async () => {
+    mockRmqClient = {
+      emit: jest.fn().mockReturnThis(),
+      subscribe: jest.fn(),
+    } as any;
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        KeywordScrapeService,
+        { provide: SCRAPING_DONE, useValue: mockRmqClient },
+      ],
+    }).compile();
+
+    service = module.get<KeywordScrapeService>(KeywordScrapeService);
+  });
+
+  it('should scrape keyword successfully', async () => {
```

Comment:
this test case is for the `scrapeKeyword()` function, it could be wrapped under `describe` to be consistent with other tests :) 
```suggestion
  describe('scrapeKeyword', () => {
    it('should scrape keyword successfully', async () => {